tests/snowflake/test_snowflake.py

```diff
@@ -869,3 +869,64 @@ def test_get_model_exception(
         snowflake_connector.get_model()
     mocked_common_get_databases.assert_called_once()
     cm.force_clean()
+
+
+@patch('snowflake.connector.connect', return_value=SnowflakeConnection)
+@patch('toucan_connectors.connection_manager.ConnectionBO.exec_close', return_value=True)
+@patch('toucan_connectors.connection_manager.ConnectionBO.exec_alive', return_value=True)
+def test_get_model_db_name_provided(
+    is_closed, close, connect, mocker, snowflake_datasource, snowflake_connector
+):
+    cm = SnowflakeConnector.get_snowflake_connection_manager()
+    mocked_common_get_databases = mocker.patch(
+        'toucan_connectors.snowflake.snowflake_connector.SnowflakeCommon.get_databases',
+        return_value=['booo'],
+    )
+    mocked_common_get_db_content = mocker.patch(
+        'toucan_connectors.snowflake.snowflake_connector.SnowflakeCommon.get_db_content',
+        return_value=pd.DataFrame(
+            [
+                {
+                    'DATABASE': 'SNOWFLAKE_SAMPLE_DATA',
+                    'SCHEMA': 'TPCH_SF1000',
+                    'TYPE': 'table',
+                    'NAME': 'REGION',
+                    'COLUMNS': '[\n  {\n    "name": "R_COMMENT",\n    "type": "TEXT"\n  },\n  {\n    "name": '
+                    '"R_COMMENT",\n    "type": "TEXT"\n  },\n  {\n    "name": "R_NAME",\n    "type": '
+                    '"TEXT"\n  },\n  {\n    "name": "R_REGIONKEY",\n    "type": "NUMBER"\n  },\n  {\n    '
+                    '"name": "R_REGIONKEY",\n    "type": "NUMBER"\n  },\n  {\n    "name": "R_NAME",'
+                    '\n    "type": "TEXT"\n  },\n  {\n    "name": "R_COMMENT",\n    "type": "TEXT"\n  },'
+                    '\n  {\n    "name": "R_NAME",\n    "type": "TEXT"\n  },\n  {\n    "name": "R_NAME",'
+                    '\n    "type": "TEXT"\n  },\n  {\n    "name": "R_REGIONKEY",\n    "type": "NUMBER"\n  '
+                    '},\n  {\n    "name": "R_COMMENT",\n    "type": "TEXT"\n  },\n  {\n    "name": '
+                    '"R_REGIONKEY",\n    "type": "NUMBER"\n  }\n]',
```

Comment:
2 columns TEXT and NUMBER could be enough since it's just a mocking test...  but, it's not a blocking point.